### PR TITLE
[LM-10577] Chatroom Topic

### PR DESCRIPTION
**Pull request by arnavr15**
## Description

**Features**
- Added Chatroom topic arch revamp

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: likeminds-chat-reactnative-integration/ChatSX/components/MessageList/styles.ts [hunk 1]
```diff
@@ -100,52 +125,54 @@ export const styles = StyleSheet.create({
...
     alignItems: "center",
   },
   reactionModalView: {
-    marginTop: Platform.OS === "ios" ? 45 : 10,
+    marginTop:
+      Platform.OS === "ios" ? Layout.normalize(45) : Layout.normalize(10),
     backgroundColor: "white",
-    borderRadius: 8,
+    borderRadius: Layout.normalize(8),
     // width: 200,
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/components/MessageList/styles.ts:153]: remove unnecessary comments

File: likeminds-chat-reactnative-integration/ChatSX/components/MessageList/styles.ts [hunk 2]
```diff
@@ -100,52 +125,54 @@ export const styles = StyleSheet.create({
...
+    padding: Layout.normalize(5),
+    shadowOpacity: Layout.normalize(0.25),
+    shadowRadius: Layout.normalize(4),
+    elevation: Layout.normalize(5),
     display: "flex",
     flexDirection: "row",
   },
   emojiModalView: {
     // marginTop: Platform.OS === 'ios' ? 45 : 10,
   },
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/components/MessageList/styles.ts:168]: remove unnecessary styles

File: likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx [hunk 3]
```diff
@@ -0,0 +1,2107 @@
...
+
+  const handleModalClose = () => {
+    setModalVisible(false);
+  };
+
+  const handleReactionModalClose = () => {
+    setIsReact(false);
+  };
+
+  const leaveChatroom = async () => {
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx:929]: add comments on every method of this file

File: example/screens/Chatroom/index.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import {
+  ChatRoom,
+  ChatroomHeader,
+  MessageList,
+  MessageInput,
+} from '@likeminds.community/chat-rn-core';
+
+const ChatroomScreen = ({navigation, route}: any) => {
```
Comment (jaixchaudhary) [inline example/screens/Chatroom/index.tsx:9]: change the type

File: likeminds-chat-reactnative-integration/ChatSX/customModals/ChatroomActionModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                      val?.id === ChatroomActions.LEAVE_CHATROOM ||
+                      val?.id === ChatroomActions.LEAVE_SECRET_CHATROOM
+                    ) {
+                      // showWarningModal();
+                      setModalVisible(false);
+                    } else if (val?.id === ChatroomActions.JOIN_CHATROOM) {
+                      if (!isSecret) {
+                        // joinChatroom();
+                      }
+                      setModalVisible(false);
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/customModals/ChatroomActionModal/index.tsx:61]: can we put some sort of customizations, to show and hide three dot menu options?

File: example/screens/Chatroom/index.tsx (hunk 4 above)
Comment (arnavr15) [reply example/screens/Chatroom/index.tsx:9]: this has been removed in new PRs

File: likeminds-chat-reactnative-integration/ChatSX/customModals/ChatroomActionModal/index.tsx (hunk 5 above)
Comment (arnavr15) [reply likeminds-chat-reactnative-integration/ChatSX/customModals/ChatroomActionModal/index.tsx:61]: will cover it in callBacks customisations PR

**Review by jaixchaudhary — APPROVED** [review]

**Pull request merged**